src/screens/Exchange/Exchange.js

```diff
@@ -839,13 +839,15 @@ class ExchangeScreen extends React.Component<Props, State> {
 
     const isFiat = isFiatProvider(offerProvider);
 
+    const disableFiatExchange = isFiat && (minOrMaxNeeded || !!offerRestricted);
+
     return (
       <ShadowedCard
         wrapperStyle={{ marginBottom: 10 }}
         contentWrapperStyle={{ paddingHorizontal: 16, paddingVertical: 6 }}
       >
         <CardWrapper
-          disabled={isTakeButtonDisabled || disableNonFiatExchange}
+          disabled={isTakeButtonDisabled || disableNonFiatExchange || disableFiatExchange}
```

Comment:
Hm, what do you think of adding single variable for disabled? This part confuses a bit `... disableNonFiatExchange || disableFiatExchange`.